src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -996,6 +992,32 @@ public int Count
         }
 
         /// <summary>
+        /// Gets the number of key/value pairs contained in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>. Should only be used after all locks
+        /// have been aquired.
+        /// </summary>
+        /// <exception cref="T:System.OverflowException">The dictionary contains too many
+        /// elements.</exception>
+        /// <value>The number of key/value pairs contained in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>.</value>
+        /// <remarks>Count has snapshot semantics and represents the number of items in the <see
+        /// cref="ConcurrentDictionary{TKey,TValue}"/>
+        /// at the moment when Count was accessed.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I'm generally not a fan of using AggressiveInlining in code like this as it asserts control over what should be a runtime-level decision. If the benefits are huge then it can be admissible though.